### PR TITLE
Object type determination with strict equivalence operator

### PR DESCRIPTION
**Pull request by nojiritakeshi**
Sorry it's a really small correction, but I made the correction.

### CODE REVIEW

File: packages/core/src/utils/object.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export function flattenObject<const T, T2 extends NestedObject<T>>(
     if (!Object.hasOwn(object, key)) continue;
     const value = object[key];
 
-    if (typeof value == "object" && value !== null) {
+    if (typeof value === "object" && value !== null) {
       const _object = flattenObject(value as NestedObject);
       for (const _key in _object) {
         if (!Object.hasOwn(_object, _key)) continue;
```
Comment (nojiritakeshi) [inline packages/core/src/utils/object.ts:35]: My question is, is this process necessary?
If it seems unnecessary, I will remove it and commit.

File: .changeset/green-gifts-deliver.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@kuma-ui/core": major
```
Comment (poteboy) [inline .changeset/green-gifts-deliver.md:2]:
```suggestion
"@kuma-ui/core": patch
```

File: packages/core/src/utils/object.ts (hunk 1 above)
Comment (poteboy) [reply packages/core/src/utils/object.ts:35]: @taishinaritomi 
IIRC I believe @taishinaritomi  made this change, can you confirm?

Comment (taishinaritomi) [reply packages/core/src/utils/object.ts:35]: It ensures that we're only processing the object's own properties and not the properties from its prototype chain.

スクリーンショット 2023-09-28 10 33 17

スクリーンショット 2023-09-28 10 33 28

Comment (nojiritakeshi) [reply packages/core/src/utils/object.ts:35]: @taishinaritomi 
I see.... .very informative. thank you...!

**Review by poteboy — APPROVED** [review]
LGTM!

**Pull request merged**